doc/policy/mempool-replacements.md

```diff
@@ -0,0 +1,64 @@
+# Mempool Replacements
+
+## Current Replace-by-Fee Policy
+
+A transaction conflicts with an in-mempool transaction ("directly conflicting transaction") if they
+spend one or more of the same inputs. A transaction may conflict with multiple in-mempool
+transactions.
+
+A transaction ("replacement transaction") may replace its directly conflicting
+transactions and their in-mempool descendants (together, "original transactions") if, in addition to
+passing all other consensus and policy rules, each of the following conditions are met:
+
+1. The directly conflicting transactions must all signal replaceability explicitly. A transaction is
+   signaling replaceability if any of its inputs have an nSequence number less than (0xffffffff - 1).
+
+   *Rationale*: See [BIP125
+   explanation](https://github.com/bitcoin/bips/blob/master/bip-0125.mediawiki#motivation).
+
+2. The replacement transaction may only include an unconfirmed input if that input was included in
+   one of the original transactions. An unconfirmed input spends an output from a
+   currently-unconfirmed transaction.
+
+   *Rationale*: When RBF was originally implemented, the mempool did not keep track of
+   ancestor scores yet. This rule was suggested as a temporary restriction.
+
+3. The replacement transaction must pay an absolute fee of at least the sum paid by the original
+   transactions.
+
+   *Rationale*: Only requiring the replacement transaction to have a higher feerate could allow an
+   attacker to bypass node minimum relay feerate requirements and cause the network to repeatedly
+   relay slightly smaller replacement transactions without adding any more fees.
+
+4. The additional fees (difference between absolute fee paid by the replacement transaction and the
+   sum paid by the original transactions) must pay for the replacement transaction's bandwidth at or
+   above the rate set by the node's incremental relay feerate. For example, if the incremental relay
+   feerate is 1 satoshi/vB and the replacement transaction is 500 virtual bytes total, then the
+   replacement must pay a fee at least 500 satoshis higher than the sum of the original transactions.
+
+   *Rationale*: Prevent a DoS attack where an attacker causes the network to repeatedly relay
+   transactions each paying a tiny additional amount in fees, e.g. just 1 satoshi.
+
+5. The number of original transactions must not exceed 100. More precisely, the sum of all
+   directly conflicting transactions' descendant counts (number of transactions inclusive of itself
+   and its descendants) must not exceed 100.
+
+   *Rationale*: Prevent a DoS attack where an attacker is able to easily occupy and flush out
+   significant portions of the node's mempool using replacements.
```

Comment:
This rule confused me a bit when I first encountered it, I think it can be worth clarifying that the attacker would replace multiple transactions at once that each have many descendants, what do you think?

```suggestion
   *Rationale*: Prevent a DoS attack where an attacker is able to easily occupy and flush out
   significant portions of the node's mempool by replacing multiple independent chains of
   transactions at once.
```

Reply:
Added!